### PR TITLE
feat(genesis): predeploy erc6551 registry

### PR DESCRIPTION
**Pull request by kingster-will**
update genesis

issue: none

### CODE REVIEW

**Review by Ramarti — COMMENTED** [review]
LGTM, but please add a test file with at least a simple test to see if `ERC6551Registry` is there and working at genesis time.

Please check /contracts/test/utils/Test.sol to understand the setup, thanks!

**Review by limengformal — APPROVED** [review]
As @Ramarti mentioned we want to add tests

**Comment by kingster-will** [comment]
Added more tests.

File: contracts/script/GenerateAlloc.s.sol [hunk 1]
```diff
@@ -332,6 +334,19 @@ contract GenerateAlloc is Script {
         console2.log("Create3 deployed at:", Predeploys.Create3);
     }
 
+    function setERC6551() internal {
+        address tmp = address(new ERC6551Registry());
```
Comment (Ramarti) [inline contracts/script/GenerateAlloc.s.sol:338]: Have we verified that `ERC6551Registry` does not set storage values in its constructor?
If it does, we have to set the same values in the correct addresses, using `vm.store`

File: contracts/src/libraries/Predeploys.sol [hunk 2]
```diff
@@ -19,6 +19,9 @@ library Predeploys {
     /// @dev We maximize compatibility with the contracts deployed by ZeframLou
     address internal constant Create3 = 0x9fBB3DF7C40Da2e5A0dE984fFE2CCB7C47cd0ABf;
 
+    /// @notice ERC6551Registry address
```
Comment (Ramarti) [inline contracts/src/libraries/Predeploys.sol:22]: Please add the reason we are not using a namespace address (0xCCCC...), for example a link to the ERC.

File: contracts/script/GenerateAlloc.s.sol (hunk 1 above)
Comment (kingster-will) [reply contracts/script/GenerateAlloc.s.sol:338]: Confirmed.  ERC6551Registry has no any storage.

File: contracts/src/libraries/Predeploys.sol (hunk 2 above)
Comment (kingster-will) [reply contracts/src/libraries/Predeploys.sol:22]: Added comments.

**Review by Ramarti — APPROVED** [review]
LGTM!

**Pull request merged**